webrtc/README.md

```diff
@@ -0,0 +1,412 @@
+
+4. _A_ creates a local offer via
+   [`RTCPeerConnection.createOffer()`](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/createOffer).
+   _A_ generates a random string and sets that string as the username (_ufrag_
+   or _username fragment_) and password on the SDP of the local offer. Finally
+   _A_ sets the modified offer via
+   [`RTCPeerConnection.setLocalDescription()`](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/setLocalDescription).
+
+   Note that this process, oftentimes referred to as "SDP munging" is disallowed
+   by the specification, but not enforced across the major browsers (Safari,
+   Firefox, Chrome) due to use-cases in the wild. See also
+   https://bugs.chromium.org/p/chromium/issues/detail?id=823036
+
+5. _A_ establishes the connection to _B_. The random string used as a _username_
+   and _password_ can be used by _B_ to identify the connection, i.e.
+   demultiplex incoming UDP datagrams per incoming connection. _B_ uses the same
+   random string for the username and password in the STUN message from _B_ to
+   _A_.
+
+6. _B_ does not know the TLS fingerprint of _A_. _B_ upgrades the incoming
+   connection from _A_ as an _insecure_ connection, learning _A_'s TLS
+   fingerprint through the WebRTC DTLS handshake. At this point the DTLS
+   handshake provides confidentiality and integrity but not authenticity.
+
+7. Messages on an `RTCDataChannel` are framed using the message framing
+   mechanism described in [Multiplexing](#multiplexing).
+
+8. The remote is authenticated via an additional Noise handshake. See
+   [Connection Security](#connection-security).
+
+#### Open Questions
+
+- Is the fact that the server accepts STUN messages from the client prone to
+  attacks?
+
+    - Can an attacker launch an **amplification attack** with the STUN endpoint
+      of the server?
+
+      The QUIC protocol defends against amplification attacks by requiring:
+
+      > an endpoint MUST limit the amount of data it sends to the unvalidated
+      > address to three times the amount of data received from that address.
+
+      https://datatracker.ietf.org/doc/html/rfc9000#section-8
+
+      For WebRTC in libp2p one could require the client (_A_) to add additional
+      bytes to its STUN message, e.g. in the STUN username and password, thus
+      making an amplification attack less attractive.
+
+    - Can a client run a **DOS attack** by sending many STUN messages with
+      different ufrags using different UDP source ports, forcing the server to
+      allocate a new peer connection for each? Would rate limiting suffice to
+      defend against this attack?
+
+  See also:
+  - https://datatracker.ietf.org/doc/html/rfc5389#section-16.2.1
+  - https://datatracker.ietf.org/doc/html/rfc5389#section-16.1.2
+
+- Do the major WebRTC server implementations support using the same UDP port for
```

Comment:
This seems to have been resolved. I think this should be moved as a statement to the section above (“major WebRTC implementations support demultiplexing…”), maybe with a short explanation how the demultiplexing is achieved.

Reply:
Removed with https://github.com/libp2p/specs/pull/412/commits/67e12daf61132fb56705f35ebd0104722d70c782 including a short explainer on how to do multiplexing on the UDP socket. Happy for additional wording suggestions.